File: app/internal/publisher/template/go/activation.go
package wsd

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

type ActivationDialog struct {
	engine      *LicenseEngine
	productName string
}

func NewActivationDialog(engine *LicenseEngine, productName string) *ActivationDialog {
	return &ActivationDialog{
		engine:      engine,
		productName: productName,
	}
}

func (d *ActivationDialog) Show() map[string]interface{} {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println(strings.Repeat("=", 60))
	fmt.Println("UNIVERSAL LICENSE ACTIVATION")
	if d.productName != "" {
		fmt.Printf("Product: %s\n", d.productName)
	}
	fmt.Println(strings.Repeat("=", 60))
	hwID := d.engine.GetHardwareID()
	fmt.Printf("Hardware ID: %s\n", hwID)
	fmt.Println()
	fmt.Print("Enter license key: ")
	key, _ := reader.ReadString('\n')
	key = strings.TrimSpace(key)
	if key == "" {
		fmt.Println("No license key entered. Activation cancelled.")
		return map[string]interface{}{
			"activated":   false,
			"cancelled":   true,
			"license_key": nil,
		}
	}
	fmt.Println("Activating license...")
	result, err := d.engine.Activate(key)
	if err != nil {
		fmt.Printf("Activation failed: %s\n", err.Error())
		return map[string]interface{}{
			"activated":   false,
			"cancelled":   false,
			"license_key": key,
		}
	}
	if success, ok := result["success"].(bool); ok && success {
		fmt.Println("License activated successfully!")
		return map[string]interface{}{
			"activated":   true,
			"cancelled":   false,
			"license_key": key,
		}
	}
	msg, _ := result["message"].(string)
	if msg == "" {
		msg, _ = result["error"].(string)
	}
	if msg == "" {
		msg = "Activation failed"
	}
	fmt.Printf("Activation failed: %s\n", msg)
	return map[string]interface{}{
		"activated":   false,
		"cancelled":   false,
		"license_key": key,
	}
}

File: app/internal/publisher/template/rust/src/activation.rs
use std::io::{self, BufRead, Write};

use crate::license_engine::LicenseEngine;

pub struct ActivationDialog;

impl ActivationDialog {
    pub fn show(engine: &mut LicenseEngine) -> Result<ActivationResult, String> {
        let stdin = io::stdin();
        let mut stdout = io::stdout();

        let _ = writeln!(stdout, "{}", "=".repeat(60));
        let _ = writeln!(stdout, "UNIVERSAL LICENSE ACTIVATION");
        if !engine.product_name.is_empty() {
            let _ = writeln!(stdout, "Product: {}", engine.product_name);
        }
        let _ = writeln!(stdout, "{}", "=".repeat(60));

        let hw_id = engine.get_hardware_id();
        let _ = writeln!(stdout, "Hardware ID: {}", hw_id);
        let _ = writeln!(stdout);

        let _ = write!(stdout, "Enter license key: ");
        let _ = stdout.flush();

        let mut key = String::new();
        stdin.lock().read_line(&mut key).map_err(|e| e.to_string())?;
        let key = key.trim().to_string();

        if key.is_empty() {
            let _ = writeln!(stdout, "No license key entered. Activation cancelled.");
            return Ok(ActivationResult {
                activated: false,
                cancelled: true,
                license_key: None,
            });
        }

        let _ = writeln!(stdout, "Activating license...");
        let _ = stdout.flush();

        match engine.activate(&key) {
            Ok(result) => {
                if result.get("success").and_then(|v| v.as_bool()).unwrap_or(false) {
                    let _ = writeln!(stdout, "License activated successfully!");
                    Ok(ActivationResult {
                        activated: true,
                        cancelled: false,
                        license_key: Some(key),
                    })
                } else {
                    let msg = result.get("message")
                        .or_else(|| result.get("error"))
                        .and_then(|v| v.as_str())
                        .unwrap_or("Activation failed");
                    let _ = writeln!(stdout, "Activation failed: {}", msg);
                    Ok(ActivationResult {
                        activated: false,
                        cancelled: false,
                        license_key: Some(key),
                    })
                }
            }
            Err(e) => {
                let _ = writeln!(stdout, "Activation failed: {}", e);
                Ok(ActivationResult {
                    activated: false,
                    cancelled: false,
                    license_key: Some(key),
                })
            }
        }
    }
}

pub struct ActivationResult {
    pub activated: bool,
    pub cancelled: bool,
    pub license_key: Option<String>,
}
